internal/corazawaf/body_buffer.go

```diff
@@ -99,30 +100,45 @@ type bodyBufferReader struct {
 }
 
 func (b *bodyBufferReader) Read(p []byte) (n int, err error) {
+	if b.br == nil {
```

Comment:
Instead of having to keep track of readers in the buffer, with some small overhead for the array, can it just have a `closed` flag which is checked here?

Reply:
I though about it but the bodybuffer is going to be reused so marking it as closed might not help. Of course we can remove that flag when restarting the transaction but maybe that is too much for the transaction to know about the body buffer

Reply:
Ah ok SGTM